api/cluster.go

```diff
@@ -415,35 +415,30 @@ func (s *Server) peerQuerySpeculativeChan(ctx context.Context, data cluster.Trac
 		reqCtx, cancel := context.WithCancel(ctx)
 		defer cancel()
 
-		originalPeers := make(map[string]struct{}, len(peerGroups))
-		receivedResponses := make(map[int32]struct{}, len(peerGroups))
-
-		responses := make(chan struct {
+		type response struct {
 			shardGroup int32
 			data       PeerResponse
 			err        error
-		}, 1)
+		}
+
+		responses := make(chan response, len(peerGroups))
+		originalPeers := make(map[string]struct{}, len(peerGroups))
+		receivedResponses := make(map[int32]struct{}, len(peerGroups))
 
 		askPeer := func(shardGroup int32, peer cluster.Node) {
 			log.Debugf("HTTP Render querying %s%s", peer.GetName(), path)
 			buf, err := peer.Post(reqCtx, name, path, data)
+			if err != nil {
```

Comment:
If the context was canceled by another goroutine, err will be non-nil.  I dont think we want to log that.

Reply:
I'm looking at: https://github.com/grafana/metrictank/blob/master/cluster/node.go#L257
and i can't see how the returned error would be non-nil in case of a cancelled context, or am I looking at the wrong thing?

Reply:
looks like you are right.  i just assumed peer.Post was like http.Post.